### PR TITLE
Stop applying server lint rules to focalbard repo.

### PR DESCRIPTION
**Pull request by pfltdv**
#### Summary
Whenever we execute golangci-lint it activates rules of mattermost-server at focalboard repo.
We should not apply server lint rules to that repo. PR contains changes to execute lint on focalboard repo with it's own rules.

#### Ticket Link
https://mattermost.atlassian.net/browse/CLD-3800

#### Release Note
```release-note
NONE
```

### CODE REVIEW

**Comment by pfltdv** [comment]
TBS pipeline run https://app.circleci.com/pipelines/github/mattermost/enterprise/17809/workflows/df9cee93-3859-47c1-bfbf-e1f1b206f627

**Review by phoinixgrr — APPROVED** [review]

File: Makefile [hunk 1]
```diff
@@ -184,7 +184,6 @@ endif
 # Prepare optional Boards build.
 BOARDS_PACKAGES=$(shell $(GO) list $(BUILD_BOARDS_DIR)/server/...)
 ifeq ($(BUILD_BOARDS),true)
-    ALL_PACKAGES += $(BOARDS_PACKAGES)
```
Comment (pfltdv) [inline Makefile:187]: If we include board packages in Allpackages , test will try to execute those tests which needs `-tag 'json1'`. How ever whenever we add that flag mattermost-server builds and tests are failing with unsupported tag error.

Comment (agnivade) [reply Makefile:187]: Let's add a comment to explain why we aren't doing this so that people are aware (possibly along with a ticket reference)

**Review by agnivade — APPROVED** [review]
Thanks Mustafa!

File: Makefile (hunk 1 above)
Comment (pfltdv) [reply Makefile:187]: Comment is added with PR and ticket reference.

**Pull request merged**